server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/SessionRoutes.scala

```diff
@@ -0,0 +1,71 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.http
+
+import java.util.function.BiFunction
+
+import io.netty.handler.codec.http.HttpHeaderNames.CONTENT_TYPE
+import io.netty.handler.codec.http.HttpResponseStatus.OK
+import javax.inject.Inject
+import org.apache.james.jmap.HttpConstants.JSON_CONTENT_TYPE_UTF8
+import org.apache.james.jmap.exceptions.UnauthorizedException
+import org.apache.james.jmap.http.SessionRoutes.JMAP_SESSION
+import org.apache.james.jmap.model.Session
+import org.apache.james.jmap.{JMAPRoutes, Serializer}
+import org.reactivestreams.Publisher
+import org.slf4j.{Logger, LoggerFactory}
+import reactor.core.scala.publisher.SMono
+import reactor.core.scheduler.Schedulers
+import reactor.netty.http.server.{HttpServerRequest, HttpServerResponse, HttpServerRoutes}
+
+object SessionRoutes {
+  private val JMAP_SESSION = "/jmap/session"
+  private val LOGGER = LoggerFactory.getLogger(classOf[SessionRoutes])
+}
+
+@Inject
+class SessionRoutes(val authFilter: AuthenticationReactiveFilter,
+                    val serializer: Serializer = new Serializer) extends JMAPRoutes {
+
+  val logger: Logger = SessionRoutes.LOGGER
+  val generateSession: BiFunction[HttpServerRequest, HttpServerResponse, Publisher[Void]] =
+    (request, response) => SMono.fromPublisher(authFilter.authenticate(request))
+      .map(_.getUser)
+      .map(SessionSupplier.generate)
+      .flatMap(session => sendRespond(session, response))
+      .onErrorResume(throwable => errorHandling(throwable, response))
+      .subscribeOn(Schedulers.elastic())
```

Comment:
why do you need to call subscribeOn here?

Reply:
it originally came from reactor-netty POC. For me I don't see any possible place to set the scheduler except this place. Need some confirmation from the author, @chibenwa @Arsnael can you help?

Reply:
Because we don't want reactor to run things on the netty executors

Reply:
I can't find any sample here https://projectreactor.io/docs/netty/release/reference/index.html#http-server that uses `subscribeOn`. Don't you think it's handled by reactor netty? Have you any evidence?

Reply:
I had a look at reactor-netty source code, there is even no reactor Scheduler import, looks like it's using some native netty http connection pools. The maximum number of connection is limited by: 
```java
reactor.netty.resourcesConnectionProvider.DEFAULT_POOL_MAX_CONNECTIONS = 
Integer.parseInt(System.getProperty(ReactorNetty.POOL_MAX_CONNECTIONS,
			"" + Math.max(Runtime.getRuntime()
			            .availableProcessors(), 8) * 2));
```

I guess we should not make it busy. All are my guess, no real evidence found

Reply:
a connection pool is not a thread pool

Reply:
Okay, but come back to your question, then the answer is: reactor-netty doesn't manage any Scheduler to do its jobs. Should or should not use `subcribeOn()` ? I don't know, maybe performance tests can tell

Reply:
I guess a single call with debug logging turned on (with threading information) would be enough to understand how it is executed

Reply:
that is `reactor-http-epoll-1` . I traced, it leads to netty `NioEventLoopGroup(workerCount, threadFactory)`